services/rollout-service/pkg/argo/argo.go

```diff
@@ -174,11 +175,14 @@ func (a ArgoAppProcessor) CreateOrUpdateApp(ctx context.Context, overview *api.G
 			}
 
 			//exhaustruct:ignore
-			emptyAppSpec := v1alpha1.ApplicationSpec{}
+			emptyAppSpec := v1alpha1.ApplicationSpec{
+				Destination: v1alpha1.ApplicationDestination{},
+				SyncPolicy:  &v1alpha1.SyncPolicy{},
+			}
 			//We have to exclude the unexported type destination and the syncPolicy
 			diff := cmp.Diff(appUpdateRequest.Application.Spec, existingApp.Spec,
 				cmp.AllowUnexported(emptyAppSpec.Destination),
-				cmp.AllowUnexported(emptyAppSpec.SyncPolicy))
```

Comment:
How about:

```go
				cmpopts.IgnoreUnexported(v1alpha1.ApplicationDestination{}),
				cmpopts.IgnoreUnexported(v1alpha1.SyncPolicy{}))
```

Reply:
I found that `IgnoreTypes` is exactly what we want. ``IgnoreUnexported` does not guarantee the diff to be excluded 100% of the times